base/intfuncs.jl

```diff
@@ -483,7 +503,41 @@ function hex(x::Unsigned, pad::Int, neg::Bool)
     String(a)
 end
 
-num2hex(n::Integer) = hex(n, sizeof(n)*2)
+"""
+    num2hex(f)
+
+An hexadecimal string of the binary representation of a number.
+See also the [`bits`](@ref) function, which is similar but gives
+a binary string, and [`hex2num`] which does the opposite conversion.
+
+```jldoctest
+julia> num2hex(Int64(4))
+"0000000000000004"
+
+julia> num2hex(2.2)
+"400199999999999a"
+
```

Comment:
excess blank line here

I think "a hexadecimal" is more common than "an hexadecimal"